pytorch_lightning/loggers/tensorboard.py

```diff
@@ -75,6 +78,7 @@ def __init__(
         save_dir: str,
         name: Optional[str] = "default",
         version: Optional[Union[int, str]] = None,
+        sub_dir: Optional[str] = None,
```

Comment:
Not sure about this, but we may want to add this as the last argument here, since otherwise it may be breaking for people using positional arguments for init.

Reply:
It really depends on what to prioritize, I prefer this order because it makes sense semantically - i.e. the order  corresponds to the actual logging path: `save_dir / name / version / sub_dir`. However, if making sure that nothing breaks is a top priority for PL then I will edit the PR.

Reply:
Yes, we are very careful with ordering here.

Reply:
clear, position has been swapped in latest commit